zipline/pipeline/loaders/utils.py

```diff
@@ -2,150 +2,85 @@
 
 import numpy as np
 import pandas as pd
-from six import iteritems
-from six.moves import zip
-
-from zipline.utils.numpy_utils import categorical_dtype, NaTns
 from zipline.utils.pandas_utils import mask_between_time
 
 
-def next_event_frame(events_by_sid,
-                     dates,
-                     missing_value,
-                     field_dtype,
-                     event_date_field_name,
-                     return_field_name):
+def is_sorted_ascending(a):
+    return (np.fmax.accumulate(a) <= a).all()
+
+
+def next_event_indexer(all_dates,
+                       all_sids,
+                       event_dates,
+                       event_timestamps,
+                       event_sids):
     """
-    Make a DataFrame representing the simulated next known dates or values
-    for an event.
+    Construct an indexer that will convert a column of event data into a 2D
+    array containing a value for the most recent next-known event for each sid
+    on each date.
```

Comment:
> I would probably clarify that the array contains an index into the event data column ('value' makes it sound like it contains the actual value from the column). 

The interpretation in the parenthetical is the correct one here.  The sentence here is saying that we're going to construct an index array that, when applied to a column of values, produces a 2D array of the values associated with the next event for each sid at each moment in time.

I'll probably replace what's here with what I just wrote above.